wandb/integration/keras_core/model_checkpoint.py

```diff
@@ -0,0 +1,173 @@
+        verbose: (int) Verbosity mode, 0 or 1. Mode 0 is silent, and mode 1
+            displays messages when the callback takes an action.
+        save_best_only: (bool) if `save_best_only=True`, it only saves when the model
+            is considered the "best" and the latest best model according to the
+            quantity monitored will not be overwritten. If `filepath` doesn't contain
+            formatting options like `{epoch}` then `filepath` will be overwritten by
+            each new better model locally. The model logged as an artifact will still be
+            associated with the correct `monitor`.  Artifacts will be uploaded
+            continuously and versioned separately as a new best model is found.
+        save_weights_only: (bool) if True, then only the model's weights will be saved.
+        mode: (Mode) one of {'auto', 'min', 'max'}. For `val_acc`, this should be `max`,
+            for `val_loss` this should be `min`, etc.
+        save_freq: (Union[SaveStrategy, int]) `epoch` or integer. When using `'epoch'`,
+            the callback saves the model after each epoch. When using an integer, the
+            callback saves the model at end of this many batches.
+            Note that when monitoring validation metrics such as `val_acc` or `val_loss`,
+            save_freq must be set to "epoch" as those metrics are only available at the
+            end of an epoch.
+        initial_value_threshold: (Optional[float]) Floating point initial "best" value of
+            the metric to be monitored.
+        artifact_type: (Optional[str]) Type of the artifact to be logged. It is set to
+            `"model"` if `save_weights_only` is set to `False` and `"weights"` otherwise
+            by default.
+    """
+
+    def __init__(
+        self,
+        filepath: StrPath,
+        monitor: str = "val_loss",
+        verbose: int = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: Mode = "auto",
+        save_freq: Union[SaveStrategy, int] = "epoch",
+        initial_value_threshold: Optional[float] = None,
+        artifact_type: Optional[str] = None,
+    ):
+        if wandb.run is None:
+            raise wandb.Error(
+                "You must call `wandb.init()` before `WandbModelCheckpoint()`"
+            )
+        with telemetry.context(run=wandb.run) as tel:
+            tel.feature.keras_model_checkpoint = True
+
+        if artifact_type is None:
+            self.artifact_type = "model-weights" if save_weights_only else "model"
+        else:
+            self.artifact_type = artifact_type
+
+        super().__init__(
+            filepath,
+            monitor,
+            verbose,
+            save_best_only,
+            save_weights_only,
+            mode,
+            save_freq,
+            initial_value_threshold,
+        )
+
+    def _is_best_checkpoint(self, logs: Optional[Dict] = None):
```

Comment:
Hey, I checked this logic properly now and realised it will not do the intended. The `self.best` is updated first in the base class when `_save_model` is called: https://github.com/keras-team/keras-core/blob/3a464f904dc8dd09de9c51d248a7fe7c575d7f3e/keras_core/callbacks/model_checkpoint.py#L242C25-L242C25 

We will have to maintain the best metric locally - initialised in the __init__ method (`self._local_best = self.best`) and pass `self._local_best` to `self.monitor_op`.